pkg/maps/nodemap/node_map.go

```diff
@@ -20,25 +20,37 @@ const (
 	MaxEntries = 16384
 )
 
-var (
-	nodeMap     *Map
-	nodeMapInit = &sync.Once{}
-)
+// Map provides access to the eBPF map node.
+type Map interface {
+	// Update inserts or updates the node map object associated with the provided
+	// IP and node id.
+	Update(ip net.IP, nodeID uint16) error
+
+	// Delete deletes the node map object associated with the provided
+	// IP.
+	Delete(ip net.IP) error
 
-// SetNodeMap sets the node map. Only used for testing.
-func SetNodeMap(m *Map) {
-	nodeMap = m
+	// IterateWithCallback iterates through all the keys/values of a node map,
+	// passing each key/value pair to the cb callback.
+	IterateWithCallback(cb NodeIterateCallback) error
 }
 
-func NodeMap() *Map {
-	nodeMapInit.Do(func() {
-		nodeMap = NewNodeMap(MapName)
-	})
-	return nodeMap
+type nodeMap struct {
+	bpfMap *ebpf.Map
```

Comment:
nit: This could have renamed unnamed as well? Or is there value in having to call bpf Map functions explicitly?

Reply:
i kind of prefer having the inner maps functions not exposed. kind of wrapping the functionality and not extending it as it would be when having it as anonymous field.